### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by KIsmoilov**
**User index page:**

- I can see the username of all other users.
- I can see the profile picture for each user.
- I can see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.

**User show page:**

- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see the user's bio.
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
- When I click to see all posts, it redirects me to the user's post's index page.

**User post index page:**

- I can see the user's profile picture.
- I can see the user's username.
- I can see the number of posts the user has written.
- I can see a post's title.
- I can see some of the post's body.
- I can see the first comments on a post.
- I can see how many comments a post has.
- I can see how many likes a post has.
- I can see a section for pagination if there are more posts than fit on the view.
- When I click on a post, it redirects me to that post's show page.

**Post show page:**

- I can see the post's title.
- I can see who wrote the post.
- I can see how many comments it has.
- I can see how many likes it has.
- I can see the post body.
- I can see the username of each commentor.
- I can see the comment each commentor left

**The N+1 problem is solved when fetching all posts and their comments for a user.**

Screen Shot 2023-03-07 at 5 55 41 PM

Screen Shot 2023-03-07 at 6 17 40 PM

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def show
-    @post = Post.where(author_id: params[:user_id], id: params[:id]).first
-    @comments = @post.comments
+    @post = Post.includes(comments: :author).find_by(author_id: params[:user_id], id: params[:id])
+    @comments = @post.comments.includes(:author)
```
Comment (danmainah) [inline app/controllers/posts_controller.rb:4]: - [ ] @KIsmoilov and @ndikumanaisaie great work fixing the N+1 problem however you didn't include a screenshot of before and after the fix, the current screenshot are for posts index and show page , kindly include screenshot of before and after fixing the N+ 1 problem

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @KIsmoilov and @ndikumanaisaie , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by KIsmoilov** [comment]
**The following are requested screenshots**

- Before
Screen Shot 2023-03-07 at 7 15 23 PM

- After
Screen Shot 2023-03-07 at 7 16 15 PM

**Review by franclobo — APPROVED** [review]
Hi @KIsmoilov and @ndikumanaisaie, 

# STATUS: APPROVED 😄 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- N + 1 problems solved ✔️ 
- All feature tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**